docs: install and running instance documentation

Exirel opened the pull request:
### Description

This is my attempt at working on #2497 with a version that I think could be good enough for Sopel 8.0.0.

I keep  it as a draft to get feedfback more than review i.e. more about what content should be or shouldn't be in here.

I took a lot from [Installing](https://sopel.chat/usage/installing/) to know what to put in here. I think the next step (here or in a future PR?) would be to add more from the [per-channel config](https://sopel.chat/usage/per-channel-configuration/) page.

What I mean is: what do you think?

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make lint` and `make test`)
- [x] I have tested the functionality of the things this change touches

Note: couldn't run `make qa` because mypy 1.6.1 fails on a line that is already in current master.

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+Sopel is designed with plugins in mind: they add features, commands, and can
+do many things to make Sopel more useful or more fun to interact with on IRC.
+
+They are two mains type of plugins you can add:
```
```suggestion
They are two main types of plugins you can add:
```

🇫🇷🇫🇷🇫🇷 😁

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+Managing plugins
+================
+
+Sopel is designed with plugins in mind: they add features, commands, and can
+do many things to make Sopel more useful or more fun to interact with on IRC.
```
```suggestion
Sopel is designed with plugins in mind: they add features/commands and can do
many things to make Sopel more useful or more fun to interact with on IRC.
```

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+
+Plugin authors might also publish their works as packages; you can find them by
+searching PyPI, or by using your favorite search engine to search for
+e.g. “sopel reminder plugin”.
```
```suggestion
e.g. "sopel reminder plugin".
```

Smart-ass text editor you've got there. 🙃

Exirel replied:
Probably a copy/paste of something else. I don't use/have smart editor replacement for quotes.

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+e.g. “sopel reminder plugin”.
+
+You can install these packages with ``pip``, the same way it is recommended to
+install Sopel. In that case, all you need is to install the plugin for the same
+Python interpreter.
```
```suggestion
install Sopel. All you need is to install the plugin for the same Python
interpreter.
```

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+
+    python3 -m pip install sopel-remind
+
+This will install the plugin from pypi where you can see `all the releases`__
```
```suggestion
This will install the plugin from PyPI where you can see `all the releases`__
```

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+install Sopel. In that case, all you need is to install the plugin for the same
+Python interpreter.
+
+For example, you can install the remind plugin from the ``sopel-remind``
```
```suggestion
For example, you can install the ``remind`` plugin from the ``sopel-remind``
```

Alternative:

```suggestion
For example, you can install the "remind" plugin from the ``sopel-remind``
```

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+
+.. note::
+
+    Each plugin can have specific steps or system dependencies you need to take
+    care of to install it. Refer to the plugin's documentation for more
+    details.
```
```suggestion
    Each plugin can have specific system dependencies or other requirements
    that you need to take care of when installing it. Refer to the plugin's
    documentation for more details.
```

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,132 @@
+plugins.
+
+You can disable a plugin by using the ``sopel-plugins disable`` command line,
+e.g. to disable the remind plugin mentionned before::
```
```suggestion
e.g. to disable the remind plugin mentioned before::
```

dgw commented on docs/source/run/install.rst:
```diff
@@ -0,0 +1,144 @@
+recommended way to install Sopel from pypi.org is to perform the following
+command::
+
+    python3 -m pip install sopel
```
The default syntax highlighting should be changed to `shell` for these files with a lot of shell command examples. They are highlighted as Python right now, and it looks a bit weird.

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+Running with systemd
+====================
+
+`systemd`__ is a service manager used by Debian and most Debian derivatives
```
```suggestion
`systemd`__ is a service manager used by Debian and most Debian-derived
```

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+====================
+
+Now that you can run the bot, you probably want it to run forever in the
+background of your system. While there are many ways to do so, there are few
```
```suggestion
background of your system. While there are many ways to do so, there are a few
```

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+* to install ``sopel`` with that user (or to change the path in the
+  ``ExecStart``)
+* to place your Sopel configuration file into ``/etc/sopel/default.cfg``, and
+  to setup its access right so that the ``sopel`` user can read and write it
```
```suggestion
  to setup its access rights so that the ``sopel`` user can read and write it
```

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+Sopel needs the following:
+
+* a non-root user to run as
+* a read and write access to its configuration file
```
```suggestion
* read/write access to its configuration file
```

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+      pid_dir = /run/sopel
+      logdir = /var/log/sopel
+
+These path are required for Sopel to know where to put its files (data, cache,
```
```suggestion
These paths are required for Sopel to know where to put its files (data, cache,
```

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+
+.. seealso::
+
+    There are more to say about systemd, its service unit file, and the
```
```suggestion
    There is more to say about systemd, its service unit files, and the
```

dgw reviewed (commented):
Once I got going it was actually hard to stop. "Just one more typo fix!"

Exirel commented:
A note here: yes, I tested the systemd's service unit myself, and it worked like a charm! I even used it with `sopel-remind` to see if it worked as intended with a plugin that create its own files from the homedir. I've to say that systemd and its automatic creation of folders with the appropriate chown & chmod is very convenient. Maybe in the future I'll consider adding a better integration with systemd, because it defines env vars that are very convenient to reuse!

SnoopJ commented:
> Note: couldn't run make qa because mypy 1.6.1 fails on a line that is already in current master.

Out of curiosity, what's the failure? I can't reproduce with that `mypy` version against either this branch or abc9322d

Exirel commented:
> Out of curiosity, what's the failure? I can't reproduce with that `mypy` version against either this branch or [abc9322](https://github.com/sopel-irc/sopel/commit/abc9322d993d53c5fa050552a0fb33a3681894a4)

```
$ mypy --check-untyped-defs sopel
sopel/lifecycle.py:147: error: Unsupported operand types for >= ("tuple[int, ...]" and "None")  [operator]
sopel/lifecycle.py:147: note: Right operand is of type "tuple[int, ...] | None"
```

Exirel commented:
We figured out, thanks to @SnoopJ intuition that it was about packaging. So here is #2535 to fix that!

Exirel commented:
> also super doesn't matter, but I'd put "running as a service" before "managing plugins" :P -- or even as just a part of the install page tbh
> but idk, that could just be me

I wondered the same thing, and I'm still on the fence. I know that I would rather have "running as a service" as its own page, so people may contribute more easily with a "here is the version for my setup that work 100% I swear".

My initial reasoning for plugin before service was that someone may want to play locally with Sopel first, install a few plugins, and see how it behaves. And then, after tinkering with it, create a proper service for their "production" environment. But that could be just me. Having the service doc first would probably invite people to consider the proper approach first.

I don't know. I'll let someone else decide (@dgw in particular).

dgw commented:
Installing Sopel as a service does make sense as a separate page for all the previously mentioned reasons, but could perhaps be a subpage of the "Install Guide" instead of a sibling.

> Might also consider tagging all your plugin repos with like `sopel-plugin` or something so it can just be a "topic" on GitHub as another method of discovery.
>
> Maybe even a "curated" list of "must-haves"

[We have this.](https://github.com/topics/sopel-plugin) The other discovery methods are cool, but no one's really _forcing_ packages or repos to be named a certain way. Plenty of plugins sit around in repos named like `<bot_nick>-modules` or some other offbeat scheme. Every suggested way we have of finding other Sopel plugins depends on the plugin author naming the package/repo a certain way, or remembering to apply a tag/topic, or submitting it to a curated-list repo (we've had [a WIP](https://github.com/sopel-irc/plugin-db/tree/prototype) for ages).

To this day I still frequently look for random plugins using a code search for "thingiwant sopel lang:python" because the most reliable way to tell if a project might be a Sopel plugin is if it imports `sopel`. 🤷‍♂️

> should push `sopel-help` instead of `sopel-remind` in the "managing plugins" page ;P

That wouldn't quite fit the tutorial flow, since `sopel-help` gets auto-installed along with `sopel`; [it's one of the dependencies](https://github.com/sopel-irc/sopel/blob/d212dd10e905fc2d90e5cf8284c24a7bbd270859/pyproject.toml#L54). `sopel-remind` is an example plugin that won't be installed by default, so the whole add-a-plugin process can be explored starting from `pip install`.

> anyway, other than the contrast on some stuff in dark mode (I flipped between light and dark a lot), I think it's very solid
> 
> and it looks much nicer than the old docs ;P

Please take any issues with the theme design to [Furo's repo](https://github.com/pradyunsg/furo), where I know the topic has previously come up from time to time. Otherwise, for Sopel-specific issues with contrast, you'll have to specify where to look. 🙂

SnoopJ commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,134 @@
+Configuring plugins
+===================
+
+A plugin can define a configuration section so you can configure its behavior.
+If the plugin supports it, you may be able to use the configuration wizard to
+edit the configuration of the plugin::
```
```suggestion
Some (but not all!) plugins define their own configuration sections so you can
configure their behavior. If a plugin supports it, you may be able to use the
configuration wizard to edit the configuration of the plugin::
```

Exirel replied:
I like your version better.

SnoopJ commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,134 @@
+
+    sopel-plugins configure remind
+
+Alternatively, you can edit your configuration file and follow the plugin's
+documentation to set the appropriate options and values:
```
```suggestion
Alternatively, you can edit your configuration file and follow the plugin's
documentation to set the appropriate options and values. Here's an
example configuration of the `sopel-remind plugin`__, which defines the
``[remote]`` section for its own use:

.. __: https://github.com/sopel-irc/sopel-remind

```

dgw replied:
Nit nit: `sopel-remind` defines the `[remind]` section for its own use, not the `[remote]` section.

Exirel replied:
I'm not sure we should add a link on "sopel-remind". Or maybe a link to the [pypi's release page](https://pypi.org/project/sopel-remind/) and not the github one, as it is a plugin that should be installed through pip.

dgw replied:
`extlinks` extension can make PyPI links easy, btw. Think <tt>:pypi:\`packagename\`</tt>.

Since the last top-level reply from @Exirel mentions needing to rebase, you'll get the newly enabled `extlinks` extension out of it as a silver lining (both the `configuration.rst` conflict and the new extension landed on `master` from #2550).

SnoopJ commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,134 @@
+    # remind's section
+    location = /path/to/remind/location
+
+When you start the bot (or restart), the bot will load the new configuration
+file. Until the bot is restarted, the modification won't take effect.
```
```suggestion
Sopel must be restarted after making changes to the configuration file.
```

SnoopJ commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+Running with systemd
+====================
+
+`systemd`__ is a service manager used by Debian and most Debian-derived
+Linux distributions. Here is an example of a service unit file to run Sopel as
```
Nit: might as well refer to Ubuntu here since it's a soft default elsewhere in this changeset

```suggestion
`systemd`__ is a service manager used by Ubuntu and most Debian-derived
Linux distributions. Here is an example of a service unit file to run Sopel as
```

SnoopJ reviewed (approved):
I had plenty of things to say, but they are mostly nits. I approve, well done!

half-duplex commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+
+.. important::
+
+    Do **not** run Sopel as the **root** user. And avoid if possible to run
```
If this isn't already the case, sopel should probably just check this and quit. Better to enforce with code than docs.

Exirel replied:
Without going into that debate: the code already does that, but this warning is less for the regular user, and more for anyone tempted to do something rather stupid. It is a warning to remind the reader that, hey, be careful out there, it's dangerous.

half-duplex commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+
+`systemd`__ is a service manager used by Debian and most Debian-derived
+Linux distributions. Here is an example of a service unit file to run Sopel as
+a service, placed at ``/etc/systemd/system/sopel.service``:
```
This doc (not necessarily this line) should at minimum mention the possibility of running it as a user service, or just give the `enable-linger`, `~/.local/share/systemd/...`, `systemctl --user enable` instructions (though the service file may need tweaks for that).

Exirel replied:
I'll let the reader do their own research on systemd. It is far more complex than what we can, or should for that matter, document here.

half-duplex commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,93 @@
+Linux distributions. Here is an example of a service unit file to run Sopel as
+a service, placed at ``/etc/systemd/system/sopel.service``:
+
+.. code-block:: systemd
```
This should link to `contrib/sopel.service` and `contrib/sopel@.service`, not duplicate them.

Exirel replied:
I think I'll remove them from contrib, that would be even better.

@dgw I feel like the contrib folder could be modified in another PR instead of this one, because @half-duplex is right about the issue of duplicating the same information.

dgw replied:
@Exirel If you're moving stuff from `contrib` into the docs, it's fair to delete the source in the same PR so as to keep the change atomic.

Exirel replied:
Alright that's fair! Thanks. :+1:

half-duplex replied:
I disagree with removing it from contrib/.
- This results in having service files in two completely unrelated places to keep in sync
- The single-instance one is no longer easily available for packagers or scripts to use, which means more users and packagers will write their own, with varying quality

dgw replied:
@half-duplex Please suggest the optimal outcome (in your opinion), then, given the concerns:

- Maintaining service files in multiple locations (in docs _and_ `contrib/`) _will_ lead to desyncs when one is inevitably forgotten while updating the other.
- The documentation should have the relevant information wholly contained within it, and not require following a link.

I share your concern about packagers/scripts losing access to templates; pointed out the same in my review note on `contrib/sopel.cfg`, though that also suffers from the desync possibility vs. examples in the docs as mentioned above.

half-duplex replied:
To me, the least bad option by a good margin is to just link the files in `contrib/`. It's better to make the user click a link than to inevitably end up with something _incorrect_ in the docs.
Alternatively, does sphinx have a way to just embed the file?

Exirel replied:
> The single-instance one is no longer easily available for packagers or scripts to use, which means more users and packagers will write their own, with varying quality

Why would we end up with something incorrect in the docs? Explain to me how is that going to happen if we have one and only one place for that information?

We don't maintain well the contrib directory, we overlooked it for most of the time. We put things in here that are mostly "do what you want with these, they are not part of our supported stuff, but sharing is cool so here it is". As a matter of fact, the single-instance unit was modified only 4 times in the last 8 years, including 2 changes unrelated to its quality: the version I rewrote is actually better now, following systemd's behavior and dedicated folders for logs, data, config, and run files (such as PID). From my perspective, it's *better* now in the doc than the version that was in contrib until now.

I'll move the multi-instance stuff into the doc, because I can get that it can be confusing to have two places for the same kind of information.

Exirel commented on docs/source/run/install.rst:
```diff
@@ -0,0 +1,143 @@
+
+.. note::
+
+    You may need to perform actions outside of Sopel's control, such as nick
+    registration. Refer to your IRC network for more information about what
+    steps you must follow to ensure you can run your bot there.
```
That's better.

Exirel commented:
Alright, there is one conflict on configuration.rst that I haven't considered yet. I can rebase to fix that.

However, I think I've fixed most of the nitpick and taken into account most of the feedbacks. There is one conversation unresolved because I'd like a different link while keeping @SnoopJ 's suggestion.

I've removed the single instance systemd service file from `contribs/` and updated the related README accordingly. I took the opportunity to add a full example of a default configuration file to connecto to Libera.

I added a bunch of links to libera, github, and created a very small FAQ section that can grow every new release (major, minor, or patch version should be fine for that!). That way there are now several references to "how to contact us" in the documentation, all pointing to the FAQ and "libera.chat, #sopel channel".

@sopel-irc/rockstars have fun reading that (again)!

dgw commented on docs/source/run.rst:
```diff
@@ -1,7 +1,26 @@
+    appreciate feedback to make it better for everyone.
+
 .. toctree::
+    :maxdepth: 2
```
Just out of curiosity, did you toy with `:titlesonly:` here?

dgw commented on docs/source/run/install.rst:
```diff
@@ -0,0 +1,139 @@
+Sopel will output log information informing you of its startup progress, such
+as loading plugins, connecting to the network, and joining channels.
+
+To stop the bot, simply exit the process (e.g. with :kbd:`Control+c`) and
```
Renders as:
<img width="145" alt="image" src="https://github.com/sopel-irc/sopel/assets/164140/16e0f52b-ad63-45cc-b616-f5cf6d402509">

I'm happy with this experiment. It looks great! 😁

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,121 @@
+.. __: https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html
+
+
+Example of configuration for libera
+===================================
```
Copying how Libera presents itself on its own site, this should be:

```suggestion
Example of configuration for Libera Chat
========================================
```

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,121 @@
+    pid_dir = /run/sopel
+    logdir = /var/log/sopel
+
+Make sure to change ``<Your Bot Nick>`` by your bot's nick, as "Sopel" is
+already taken; and set your own's nick as the owner instead of ``<Your Nick>``.
```
```suggestion
Make sure to replace ``<Your Bot Nick>`` with your bot's nick, as "Sopel" is
already taken; and set your own nick as the owner instead of ``<Your Nick>``.
```

Side note: Connecting to Libera with no `owner_account` set will make Sopel [nag the user about account matching](https://github.com/sopel-irc/sopel/blob/72493dcd4bd0b44d88a3a72f6c1948cfbd695ecd/sopel/coretasks.py#L399-L411) every time it starts, because Libera supports `account-tag`:

> This network supports using network services to identify you as my owner, rather than just matching your nickname. This is much more secure. If you'd like to do this, make sure you're logged in and reply with ".useserviceauth"

Exirel replied:
Yes I know! That's how I usually set up my bot: I run it and let it ask me to use the command. 😁

dgw replied:
That, um, definitely starts users early on learning that Sopel will clobber comments in the config file 😅

dgw reviewed (commented):
I caught what I could. I'm confident that the other reviewers will catch most of what I missed. 😁

half-duplex commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,121 @@
+* and to edit the ``[core]`` section of this configuration file to set the
+  appropriate paths:
+
+  .. code-block:: ini
+
+      [core]
+      homedir = /var/lib/sopel
+      pid_dir = /run/sopel
+      logdir = /var/log/sopel
```
We should consider adding something like `-o core.homedir=...` to `sopel start`, so the service file can override some settings there instead of requiring a specific Sopel config to work correctly.

dgw replied:
[Environment variables](https://sopel.chat/docs/cli#overriding-individual-settings) already allow this without needing to touch the CLI code.

half-duplex replied:
Ah. I don't like the `.`/`_` thing, but I'm inclined to set those vars and not ask config changes then.

Exirel commented on contrib/sopel.cfg:
```diff
@@ -1,19 +0,0 @@
-# Default sopel configuration file for Fedora
```
Given what we have in the `contrib` folder, I don't think anyone should use what it's in here directly without carefully reading the associated readme, and certainly not use an automated process to grab stuff from here.

Exirel commented:
Alright folks. It's pretty much ready to go live. This contains a pretty good next step for the "running the bot" chapter, with detailed instructions to install, configure, and run the bot, and there is even a start of a FAQ to iterate on in future PR.

I rebased to get the latest fix and doc's config modifications, and squashed everything because at this point it was just "apply review" commits. I invite you to read the preview (link in the CI's report), it looks pretty darn good to me!

SnoopJ commented:
I agree, let's get it in! The best way to get additional improvements here is to get this additional documentation in front of real users :smiling_imp:

dgw commented on contrib/README.md:
```diff
@@ -1,28 +1,5 @@
-`sopel.service` and `sopel.cfg` are designed to be distributed by third parties
-such as Fedora Project or Arch Linux.
-
-`sopel.cfg` is a default configuration file for Sopel, that assumes the OS is
```
This was wrong anyway, in hindsight. It's a description of `sopel.conf`. 😅

dgw commented on docs/source/run.rst:
```diff
@@ -1,7 +1,26 @@
 
-    cli
-    configuration
+.. __: https://github.com/sopel-irc/sopel/
```
Extremely nitpicky; GH doesn't even bother redirecting this:

```suggestion
.. __: https://github.com/sopel-irc/sopel
```

Exirel replied:
The most nitpicky of nitpick History. 😁

dgw commented on docs/source/run/install.rst:
```diff
@@ -0,0 +1,139 @@
+
+This will run a wizard that helps you create a configuration file. It will
+ask you questions to fill in the details. When you see something in ``[square
+brackets]``, that's the default setting, and you can just hit Enter to keep
```
```suggestion
brackets]``, that's the default setting, and you can just hit :kbd:`Enter` to keep
```

Only because I loved so much what `:kbd:` did elsewhere… And it looks good here too:

<img width="150" src="https://github.com/sopel-irc/sopel/assets/164140/a49e078a-b081-4473-b0f4-ea3b0d2cd7ca">

This is also very nitpicky.

Exirel replied:
> This is also very nitpicky.

But so much better!

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,137 @@
+However, you may want to enable only a set of plugins, and disable every other
+one by default.
+
+To do so, the enable command has a ``--allow-only`` option: this will add the
+plugin to the :attr:`~sopel.config.core_section.CoreSection.enable` list::
```
```suggestion
The ``enable`` command has an ``--allow-only`` option. For example, to add the
``remind`` plugin to the :attr:`~sopel.config.core_section.CoreSection.enable`
list, you would use this command line::
```

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,137 @@
+
+When you start the bot (or restart), the bot will load the plugins from the
+``enable`` list, and **only these plugins**.
+
```
Disclaimer up front: This might be too much detail. An alternative would be mentioning the `-a` switch as a shortcut to typing out the whole `--enable-only` name all the time.

```suggestion

.. note::

    You only need to use the ``--enable-only`` option once. If the ``enable``
    list is not empty, ``sopel-plugins enable`` and ``sopel-plugins disable``
    will add and remove plugins in that list.

```

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,137 @@
+Configuring plugins
+===================
+
+Some (but not all!) plugins define their own configuration sections so you can
+configure their behavior. If a plugin supports it, you may be able to use the
+configuration wizard to edit the configuration of the plugin::
```
```suggestion
Some (but not all!) plugins define their own configuration sections so you can
configure their behavior. If a plugin supports it, you may be able to use the
configuration wizard to do this::
```

(It's just simpler. Too much "configur{e|ation)" in the paragraph before. 😅)

dgw commented on docs/source/run/plugin.rst:
```diff
@@ -0,0 +1,137 @@
+    # remind's section
+    location = /path/to/remind/location
+
+Sopel must be restarted after making changes to the configuration file.
```
This is one that you should feel free to reject. I just felt while reading the deploy preview that this lonely line of text below a code-block is easy to miss, visually, and the admonition styling would help.

```suggestion
.. important::

    Sopel must be restarted after making changes to the configuration file.

```

dgw commented on docs/source/run/service.rst:
```diff
@@ -0,0 +1,174 @@
+Running as a service
+====================
+
+Now that you can run the bot, you probably want it to run forever in the
```
```suggestion
Once you have your bot set up, you will probably want it to run forever in the
```

dgw reviewed (commented):
Maybe if @SnoopJ had posted his last comment as a review instead, I wouldn't have bothered going through again. But I did, and I have a few final suggestions. (Before anyone says anything: Sometimes even _I_ hate how nitpicky I am, but it's for a good cause!)

dgw reviewed (approved):
Apparently you liked pretty much all of my nitpicks. As a reward (?) I won't go through again and risk finding more. 😜
